### DISCUSSION TITLE
How to modify the file path of the log at runtime？

### DISCUSSION THREAD

**Discussion by shanhai3000** [uber-go/zap#1296] [Q&A]
How to modify the file path of the log at runtime? I need to dynamically write logs to different files, but I know I can't initialize all logger instances
Please help

**Comment by abhinav** [comment]
@shanhai3000 This will likely require a custom `zapcore.WriteSyncer`. Zap does not provide an official means to do this. There are very likely existing high-quality implementations of rolling file rotation or similar depending on your needs that you can plug into the `WriteSyncer` interface.
